go/worker/compute/executor/committee/node.go

```diff
@@ -953,6 +960,11 @@ func (n *Node) worker() {
 			func() {
 				n.commonNode.CrossNode.Lock()
 				defer n.commonNode.CrossNode.Unlock()
+
+				if state, ok := n.state.(StateProcessingBatch); !ok || state.done != processingDoneCh {
+					n.logger.Warn("state snapshot changed unexpectedly, aborting")
```

Comment:
No need to warn here, maybe log at debug level or just remove as this is expected to happen.